packages/frontend-2/components/server-management/SettingsDialog.vue

```diff
@@ -0,0 +1,185 @@
+          v-model="company"
+          label="Owner"
+          name="owner"
+          placeholder="Owner"
+          show-label
+        />
+        <FormTextInput
+          v-model="adminContact"
+          label="Admin Email"
+          name="adminEmail"
+          placeholder="Admin Email"
+          show-label
+          :rules="emailRules"
+          :type="'email'"
+        />
+        <FormTextInput
+          v-model="termsOfService"
+          label="Url pointing to the terms of service page"
+          name="terms"
+          show-label
+        />
+        <FormCheckbox
+          v-model="inviteOnly"
+          label="Invite only mode - Only users with an invitation will be able to join"
+          name="inviteOnly"
+          show-label
+        />
+      </div>
+    </form>
+  </LayoutDialog>
+</template>
+
+<script setup lang="ts">
+import { useQuery, useMutation } from '@vue/apollo-composable'
+import { useForm } from 'vee-validate'
+import { isEmail, isRequired } from '~~/lib/common/helpers/validation'
+import { graphql } from '~~/lib/common/generated/gql'
+import { useGlobalToast, ToastNotificationType } from '~~/lib/common/composables/toast'
+import { LayoutDialog, FormTextInput, FormTextArea } from '@speckle/ui-components'
+import { useLogger } from '~~/composables/logging'
+
+type FormValues = {
+  name: string
+  description: string
+  company: string
+  adminContact: string
+  termsOfService: string
+  inviteOnly: boolean
+}
+
+const serverInfoQuery = graphql(`
+  query ServerSettingsDialogData {
+    serverInfo {
+      name
+      description
+      adminContact
+      company
+      termsOfService
+      inviteOnly
+    }
+  }
+`)
+
+const serverInfoUpdateMutation = graphql(`
+  mutation ServerInfoUpdate($info: ServerInfoUpdateInput!) {
+    serverInfoUpdate(info: $info)
+  }
+`)
+
+const props = defineProps<{
+  open: boolean
+  title: string
+}>()
+
+const emit = defineEmits<{
+  (e: 'update:open', val: boolean): void
+  (e: 'server-info-updated'): void
+}>()
+
+const dialogButtons = computed(() => [
+  {
+    text: 'Cancel',
+    props: { color: 'secondary', fullWidth: true, outline: true },
+    onClick: () => (isOpen.value = false)
+  },
+  {
+    text: 'Save',
+    props: { color: 'primary', fullWidth: true, outline: false },
+    onClick: onSubmit
+  }
+])
+
+const logger = useLogger()
+
+const { triggerNotification } = useGlobalToast()
+
+const { handleSubmit } = useForm<FormValues>()
+
+const { result } = useQuery(serverInfoQuery)
+const { mutate: updateServerInfo } = useMutation(serverInfoUpdateMutation)
+
+const name = ref('')
+const description = ref('')
+const company = ref('')
+const adminContact = ref('')
+const termsOfService = ref('')
+const inviteOnly = ref(false)
+
+const isOpen = computed({
+  get: () => props.open,
+  set: (newVal) => emit('update:open', newVal)
+})
+
+const emailRules = [isEmail]
+const requiredRule = [isRequired]
+
+const onSubmit = handleSubmit(async () => {
```

Comment:
Doesn't appear that the changes made in this mutation to `serverInfo` fields are actually refetched or manually updated in the GQL local cache. I see there's a `server-info-updated` event that is handled in the parent component, but that one only refetches a couple of `serverInfo` fields - name and version.

So I assume that after closing the dialog, if there's any component on the page that renders server info again, it will show old/stale values

Reply:
You can test this out yourself. Make a change to one of the fields that aren't refetched/manually updated in the cache like "Admin contact". Then close the dialog and open it up again - it's going to show the old/stale value.